tests/sles4sap/publiccloud/qesap_terraform.pm

```diff
@@ -95,34 +90,76 @@ sub create_hana_vars_section {
     # Cluster related setup
     my %hana_vars;
     if ($ha_enabled == 1) {
-        $hana_vars{sap_hana_install_software_directory} = get_required_var("HANA_MEDIA");
-        $hana_vars{sap_hana_install_master_password} = get_required_var("_HANA_MASTER_PW");
-        $hana_vars{sap_hana_install_sid} = get_required_var("INSTANCE_SID");
-        $hana_vars{sap_hana_install_instance_number} = get_required_var("INSTANCE_ID");
-        $hana_vars{sap_domain} = get_var("SAP_DOMAIN", "qesap.example.com");
-        $hana_vars{primary_site} = get_var("HANA_PRIMARY_SITE", "site_a");
-        $hana_vars{secondary_site} = get_var("HANA_SECONDARY_SITE", "site_b");
-        set_var("SAP_SIDADM", lc(get_var("INSTANCE_SID") . "adm"));
+        $hana_vars{sap_hana_install_software_directory} = get_required_var('HANA_MEDIA');
+        $hana_vars{sap_hana_install_master_password} = get_required_var('_HANA_MASTER_PW');
+        $hana_vars{sap_hana_install_sid} = get_required_var('INSTANCE_SID');
+        $hana_vars{sap_hana_install_instance_number} = get_required_var('INSTANCE_ID');
+        $hana_vars{sap_domain} = get_var('SAP_DOMAIN', 'qesap.example.com');
+        $hana_vars{primary_site} = get_var('HANA_PRIMARY_SITE', 'site_a');
+        $hana_vars{secondary_site} = get_var('HANA_SECONDARY_SITE', 'site_b');
+        set_var('SAP_SIDADM', lc(get_var('INSTANCE_SID') . 'adm'));
     }
     return (\%hana_vars);
 }
 
+=head2 create_instance_data
+
+    Create and populate a publiccloud::provider compatible memory structure with the current
```

Comment:
Why it says `memory` ?

Reply:
Does not make sense to move this new method to a library file, maybe an existing one? Seems a generic functionality to me.

Reply:
> Does not make sense to move this new method to a library file, maybe an existing one? Seems a generic functionality to me.

Could be. I would propose to move it whenever we need to call this from outside the module.

Reply:
`in memory` is like `an instance of the class`, right?

Reply:
> `in memory` is like `an instance of the class`, right?

More like a list of `publiccloud::instance` class instances, but yes.

Reply:
I will re-word the documentation.